src/Card/CardFooter/CardFooter.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { useResizeDetector } from 'react-resize-detector/build/withPolyfill';
+
+import { useReversedChildren } from '../../hooks';
+
+export const CardFooterContext = React.createContext();
+
+export const CARD_FOOTER_BREAKPOINTS = {
+  STACKED: 460,
+};
+
+const CardFooter = ({
+  className,
+  children,
+  stacked: controlledStacked,
+  breakpoints,
+  ...attrs
+}) => {
+  const [stacked, setStacked] = useState(controlledStacked);
+
+  const handleResize = (width) => {
+    if (controlledStacked) {
+      // ``stacked`` was passed in as a prop so this component is being used as a
+      // controlled component; rely on this prop instead of setting ``stacked`` state below.
+      return;
+    }
+
+    // TODO: what to do about hardcoded width breakpoints...?
+    if (width < (breakpoints?.stacked || CARD_FOOTER_BREAKPOINTS.STACKED)) {
+      setStacked(true);
+    } else {
+      setStacked(false);
+    }
+  };
+
+  const { ref } = useResizeDetector({
+    handleHeight: false,
+    refreshMode: 'throttle',
+    refreshRate: 100,
+    onResize: handleResize,
+  });
+
+  const reversedChildren = useReversedChildren(children);
+
+  // ``contextValue`` must be memoized to ensure the object reference stays consistent on re-renders,
+  // so that children of CardFooterContext do not constantly re-render, too.
+  const contextValue = useMemo(
+    () => ({
+      stacked,
+    }),
+    [stacked],
+  );
+
+  return (
+    <div
+      ref={ref}
+      {...attrs}
+      className={classNames('pgn__card-footer', className, {
+        stacked,
+      })}
+    >
+      <CardFooterContext.Provider value={contextValue}>
```

Comment:
My gut reaction to the CardFooter components are that they're over-engineered. What if the usage presented more like this:

```
  <Card.Footer>
    <Card.Footer.Actions>
        <Button variant="tertiary">No, thanks</Button>
        <Button as="a" href="https://edx.org" variant="primary">Learn more</Button>
    <Card.Footer.Actions>
  </Card.Footer>
```

You could control block vs inline block buttons using `cloneElement` on each child of card.footer.actions. https://reactjs.org/docs/react-api.html#cloneelement . Something similar is done in ValidationFormGroup https://github.com/edx/paragon/blob/master/src/ValidationFormGroup/index.jsx#L52. (In that use, Context may have been a better solution, but here I think Context is overkill for what needs to be done)

Reply:
Also, I think* it's better to use Button directly instead of a new component. I don't yet know why I think this and curious on your ideas here.